### PR TITLE
Unbreak CI by getting rid of golangci-lint

### PR DESCRIPTION
**Pull request by Jacalz**
### Description:

We have recently encountered golangci-lint being unreliable. This PR moves us back to the old style of CI we had and also incorporates a few more improvements on top of it (see commits). It also removes the old `.travis.yml` file that was laying around and removes the commented out modules check because Go 1.16 complains about incorrect modules either way (we got that for free) :slightly_smiling_face:

### Checklist:

- [ ] Tests included.
- [x] Lint and formatter run with no errors.
- [x] Tests all pass.

### CODE REVIEW

File: .github/workflows/static_analysis.yml [hunk 1]
```diff
@@ -6,22 +6,32 @@ jobs:
...
-        version: v1.37
-        skip-go-installation: true
+    - name: Vet
+      run: go vet -tags ci ./...
+
+    - name: Goimports
+      run: test -z $(goimports -e -d . | tee /dev/stderr)
+
+    - name: Gocyclo
+      run: gocyclo -over 50 .
```
Comment (fpabl0) [inline .github/workflows/static_analysis.yml:31]: I don't know anything about CI, but shouldn't this be 30 to match the current behavior?

Comment (Jacalz) [reply .github/workflows/static_analysis.yml:31]: You are absolutely correct. I copied over the CI from FyneDesk thinking it would be updated 😅

**Comment by andydotxyz** [comment]
Probably should remove the .golangci file as well?

**Comment by Jacalz** [comment]
> Probably should remove the .golangci file as well?

Indeed. I missed that.

**Comment by Jacalz** [comment]
I have held off upgrading the platform tests as version 1.6.09 seems to fail, see https://github.com/WillAbides/setup-go-faster/issues/13.

**Review by andydotxyz — APPROVED** [review]
Thanks, back to (slightly slower) stability :)

**Pull request merged**